### PR TITLE
Upgrade to NB-23

### PR DESCRIPTION
**Pull request by Achal1607**
This PR addresses the following updates:
- Integration of NB-23
- Removal of outdated NB-23 patches
- Upgrade of nbcode
- Enhancement of frontend components to align with NB-23 requirements

### CODE REVIEW

File: vscode/src/extension.ts [hunk 1]
```diff
@@ -415,9 +415,14 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
...
+            const params = [];
+            if (typeof template === 'string') {
+                params.push(template);
+            }
+            params.push(contextUri(ctx)?.toString(), vscode.window.activeTextEditor?.document?.uri?.toString());
+            const res = await vscode.commands.executeCommand(COMMAND_PREFIX + '.new.from.template', ...params);
+            
             if (typeof res === 'string') {
                 let newFile = vscode.Uri.parse(res as string);
                 await vscode.window.showTextDocument(newFile, { preview: false });
```
Comment (sid-srini) [inline vscode/src/extension.ts:428]: Do incorporate the fix in the function const `runDebug` on line 642 where:
```
            if (!methodName) {
                debugConfig['methodName'] = methodName;
            }
```
is fixed to:
```
            if (methodName) {
                debugConfig['methodName'] = methodName;
            }
```

**Review by sid-srini — CHANGES_REQUESTED** [review]
Thanks @Achal1607 and @lahodaj for this big task done.

I've added 1 minor comment which brings a fix done in *extension.ts*.
Otherwise, all good.

**Comment by Achal1607** [comment]
@sid-srini I have tried to address the review comments. Thanks

**Review by sid-srini — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**